web_src/js/modules/tippy.js

```diff
@@ -110,7 +123,13 @@ export function initGlobalTooltips() {
     attributeFilter: ['data-tooltip-content'],
   });
 
+  // init tooltips
   attachChildrenLazyTooltip(document.documentElement);
+
+  // init time elements
+  for (const el of document.querySelectorAll('relative-time.time-since')) {
+    swapTitleForTooltip(el);
+  }
```

Comment:
I don't think we need this hard-coded selectors.

I will use a general design for such requirement and benefit future libraries. eg:

`<span title="from original library" data-tooltip-content>`

Reply:
Done in da185ae62de1e082699cf069a5efaf8fbc26431e , no need to hard-code 'relative-time.time-since'.

As a general framework, the tooltip module shouldn't be specialized for its downstream usages, keep it independent.

Reply:
I'd be careful to not trigger useless computation during mutation events. I think the current observer triggers a few of these. See https://github.com/go-gitea/gitea/commit/da185ae62de1e082699cf069a5efaf8fbc26431e#r108257021.

Reply:
> Especially thing like `querySelectorAll` during a mutation event is very expensive, I think we may be able to do without. it could trigger hundrets of times when many elements are added, which I guess may even freeze the page momentarily.

No, no performance degradation, the code is just like before.

`childList` and `attributeFilter` are independent.

Reply:
> . it could trigger hundrets of times when many elements are added,

It won't happen. If you add new nodes, only `childList` is triggered, the `attributeFilter` won't be triggered.

Reply:
Hmm, do you think we could remove the `querySelectorAll`? As I understand it, `childList` triggers for every child down the tree, so it may be unnecessary?

Reply:
`querySelectorAll` just works like before, the behavior is not changed by this PR. It is necessary for making the tooltip work for Vue components (and other dynamically loaded contents).

And, I believe `querySelectorAll('[attribute]')`is very fast for modern browsers, there won't be any performance problem.

Reply:
ps: if we find there is any real performance problem (unlikely IMO), we can optimize it in the future.

All my framework-level code is designed as independent/transparent as possible, so each part could be optimized separately.